modules/claims_api/lib/claims_api/error/soap_error_handler.rb

```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require 'nokogiri'
+
+module ClaimsApi
+  class SoapErrorHandler
+    # list of fault codes: https://hub.verj.io/ebase/doc/SOAP_Faults.htm
+    # list faraday errors: https://lostisland.github.io/faraday/middleware/raise-error
+
+    def handle_errors(response)
+      hash = Hash.from_xml(response.body)
+
+      return if hash['Envelope']['Body']['Fault'].blank?
+
+      fault_description = get_fault_description(hash)
+
+      get_exception(fault_description)
+    end
+
+    def get_fault(hash)
+      fault_code = hash['Envelope']['Body']['Fault']['faultcode']
+      fault_code.split(':')[1]
+    end
+
+    def get_fault_description(hash)
+      fault_string = hash['Envelope']['Body']['Fault']['faultstring']
+      return {} if fault_string.include?('IntentToFileWebService') && fault_string.include?('not found')
+
+      'not_found' if fault_string.include?('could not be converted') || fault_string.include?('not found')
```

Comment:
nitpick: I'd personally put a 'return' here so it'd match ln27, then the `if` of ln30 doesn't have to run, but that's just me